edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -115,6 +116,7 @@ public static async Task<int> MainAsync(IConfiguration configuration)
                 }
 
                 edgeDeviceHostName = configuration.GetValue<string>(Constants.EdgeDeviceHostNameKey);
+                parentEdgeHostname = configuration.GetValue<string>(Constants.GatewayHostnameVariableName);
```

Comment:
when edge agent 1st started by iotedged, iotedged will inject this environment variable.